org-code-javabuilder/media/src/main/java/org/code/media/AudioUtils.java

```diff
@@ -156,4 +136,67 @@ public static void writeBytesToOutputStream(byte[] bytes, ByteArrayOutputStream
   public static int getDefaultSampleRate() {
     return DEFAULT_SAMPLE_RATE;
   }
+
+  /**
+   * Loads and reads the audio samples from the given asset file
+   *
+   * @param filename Name of the asset file
+   * @return samples
+   * @throws FileNotFoundException
+   */
+  public static double[] readSamplesFromAssetFile(String filename) throws FileNotFoundException {
+    try {
+      final URL audioFileUrl = new URL(GlobalProtocol.getInstance().generateAssetUrl(filename));
+      final AudioInputStream audioInputStream =
+          AudioUtils.convertStreamToDefaultAudioFormat(
+              AudioSystem.getAudioInputStream(audioFileUrl));
+      return AudioUtils.readSamplesFromInputStream(audioInputStream);
+    } catch (IOException e) {
+      throw new FileNotFoundException("Could not find file: " + filename);
```

Comment:
Is this a user-facing string? If so, we'll need to be able to translate it on the client side and this should be set up as a child of JavabuilderException. Take a look at the ILLEGAL_METHOD_ACCESS exception key and its use for an example of passing metadata (such as `filename`) to the client

Reply:
I think in the API we want this to throw a FileNotFoundException that students will see and can try/catch on (we have the same problem on the image side that the message is in English instead of a key). Can we make it translatable and maintain the FileNotFoundException?

Reply:
Yep, this is supposed to be same behavior as with images. I can turn this into a translatable exception for image and audio as a follow-up if that works?

Reply:
Follow-up sounds good. Mind logging a work item to track it?

Reply:
Sure thing, done: https://codedotorg.atlassian.net/browse/CSA-553